### DISCUSSION TITLE
Using a custom .tf for a specific terragrunt config

### DISCUSSION THREAD

**Discussion by denniskreibich** [gruntwork-io/knowledge-base#746] [How do I]
Hey there,
is there any possibility to use a separate .tf file with e.g. custom workload code for a specfic terragrunt.hcl?
Like:
![image](https://github.com/gruntwork-io/knowledge-base/assets/78085551/9c98c3ea-c9ab-48c0-9115-3fe2a04981e2)
That "custom code" should then be executed by that configuration, in addition to the acutal source.

Something like multiple sources would also be fine

---

<ins datetime="2023-06-30T08:25:38Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110311">Tracked in ticket #110311</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
one `terragrunt.hcl` can reference only one `terraform` block
so far I thought about a couple of approaches:
* use generate block to put custom terraform code in the respective module
* use terraform hooks to copy additional files

References:
https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#generate
https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#terraform
